### ISSUE TITLE
✅ 🔗 🐞 Bug Report: CSS breaks when updating Single templates using Additional Breakpoints (Duplicate of #16819)

### ISSUE THREAD

**Issue by maurocarreraok** [elementor/elementor#21892]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Every time a single or any type of template is changed, CSS on all templates start to break.

Here an example:

https://www.loom.com/share/cce7df8db0b14e998772c84d80b079e7

### Steps to reproduce

View video.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Source distribution v5.7.39-42
	PHP Version: 8.1.17
	PHP Memory Limit: 768M
	PHP Max Input Vars: 3000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: https://c42.studio
	Home URL: https://c42.studio
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 256M
	Max Memory limit: 768M
	Permalink Structure: /blog/%category%/%postname%/
	Language: en-US
	Timezone: -3
	Debug Mode: Inactive

== Theme ==
	Name: C42.Studio
	Version: 1.0
	Author: C42.Studio
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.6.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.11.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.11.6
		Author: Elementor.com

	Export All URLs
		Version: 4.4
		Author: Atlas Gondal

	GN Publisher
		Version: 1.5.6
		Author: Chris Andrews

	Imagify
		Version: 2.1.1
		Author: Imagify – Optimize Images & Convert WebP

	JetEngine
		Version: 3.1.3.1
		Author: Crocoblock

	JetMenu
		Version: 2.2.4
		Author: Crocoblock

	JetSearch
		Version: 3.0.3
		Author: Crocoblock

	JetSmartFilters
		Version: 3.1.0
		Author: Crocoblock

	JetTricks
		Version: 1.4.3
		Author: Crocoblock

	MainWP Child
		Version: 4.4.0.2
		Author: MainWP

	Rank Math SEO
		Version: 1.0.110.2
		Author: Rank Math

	Rank Math SEO PRO
		Version: 3.0.32
		Author: Rank Math

	Save UTMs to Elementor Form
		Version: 1.5
		Author: C42.Studio

	SiteGround Security
		Version: 1.4.1
		Author: SiteGround

	Site Kit by Google
		Version: 1.96.0
		Author: Google

	WP All Export Pro
		Version: 1.8.3
		Author: Soflyy

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast


== Elements Usage ==
	
	container : 5
		breadcrumbs : 1
		button : 8
		container : 68
		heading : 24
		image : 37
		image-carousel : 1
		jet-smart-filters-active : 1
		jet-smart-filters-checkboxes : 1
		jet-smart-filters-search : 2
		testimonial-carousel : 1
		text-editor : 1
	page : 2
		button : 1
		container : 1
		heading : 2
	wp-post : 0
		button : 23
		container : 13
		heading : 6
	wp-page : 22
		button : 50
		container : 155
		heading : 66
		image : 12
		image-carousel : 2
		jet-smart-filters-active : 1
		jet-smart-filters-radio : 1
		jet-smart-filters-search : 1
		jet-smart-filters-select : 1
		jet-unfold : 1
		lottie : 2
		shortcode : 1
		template : 4
		text-editor : 19
	single-page : 2
		button : 5
		call-to-action : 8
		container : 21
		form : 1
		heading : 10
		image : 1
		image-carousel : 1
		lottie : 2
		posts : 1
		template : 5
		text-editor : 2
		theme-post-content : 1


== Settings ==
	
	cpt_support: post, page, jet-menu, l, jet-menu
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	recaptcha_v3: Active


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Inactive
	Inline Font Icons: Active
	Accessibility Improvements: Active
	Additional Custom Breakpoints: Active
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Default to New Theme Builder: Inactive
	Hello Theme Header & Footer: Inactive
	Editor V2: Inactive by default
	Landing Pages: Inactive
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Save as Default: Active by default
	Page Transitions: Inactive
	Notes: Inactive
	Loop: Active
	Form Submissions: Active
	Scroll Snap: Inactive
	Mega Menu: Inactive by default


== Log ==
	
Log: showing 20 of 282023-03-12 15:21:08 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.3',
  'to' => '3.11.4',
)]
2023-03-12 15:21:09 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-12 15:21:09 [info] elementor::elementor_updater Started 
2023-03-12 15:21:09 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-12 15:21:09 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-12 15:21:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.3',
  'to' => '3.11.4',
)]
2023-03-12 15:21:10 [info] elementor-pro::elementor_pro_updater Started 
2023-03-12 15:21:10 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-12 15:21:10 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-12 15:21:10 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-12 15:21:11 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.3',
  'to' => '3.11.4',
)]
2023-03-12 15:21:11 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-15 11:16:18 [info] elementor-pro::elementor_pro_updater Started 
2023-03-15 11:16:18 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-15 11:16:18 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-15 11:16:18 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.5',
  'to' => '3.11.6',
)]
2023-03-15 11:16:25 [info] elementor::elementor_updater Started 
2023-03-15 11:16:25 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-15 11:16:26 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-15 11:16:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]

PHP: showing 5 of 5PHP: 2023-03-12 13:53:45 [notice X 204][/home/customer/www/c42.studio/public_html/wp-content/plugins/elementor-pro/modules/custom-css/module.php::76] trim(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-12 13:53:49 [notice X 22706][/home/customer/www/c42.studio/public_html/wp-content/plugins/elementor-pro/modules/loop-builder/module.php::200] strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-12 13:59:43 [warning X 152][/home/customer/www/c42.studio/public_html/wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/site-logo.php::32] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-12 15:03:54 [notice X 2][/home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/includes/template-library/sources/local.php::747] ob_end_clean(): Failed to delete buffer. No buffer to delete [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-15 11:15:52 [error X 1][/home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/includes/frontend.php::1399] Uncaught Error: Class "Elementor\Core\Responsive\Responsive" not found in /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/includes/frontend.php:1399
Stack trace:
#0 /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/core/base/base-object.php(142): Elementor\Frontend->get_init_settings()
#1 /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/core/base/base-object.php(40): Elementor\Core\Base\Base_Object->ensure_settings()
#2 /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/core/base/app.php(36): Elementor\Core\Base\Base_Object->get_settings()
#3 /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/includes/frontend.php(612): Elementor\Core\Base\App->print_config()
#4 /home/customer/www/c42.studio/public_html/wp-content/plugins/elementor/includes/frontend.php(803): Elementor\Frontend->enqueue_scripts()
#5 /home/customer/www/c42.studio/public_html/wp-includes/class-wp-hook.php(308): Elementor\Frontend->wp_footer('')
#6 /home/customer/www/c42.studio/public_html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#7 /home/customer/www/c42.studio/public_html/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#8 /home/customer/www/c42.studio/public_html/wp-includes/general-template.php(3059): do_action('wp_footer')
#9 /home/customer/www/c42.studio/public_html/wp-content/themes/hello-elementor/footer.php(23): wp_footer()
#10 /home/customer/www/c42.studio/public_html/wp-includes/template.php(783): require_once('/home/customer/...')
#11 /home/customer/www/c42.studio/public_html/wp-includes/template.php(718): load_template('/home/customer/...', true, Array)
#12 /home/customer/www/c42.studio/public_html/wp-includes/general-template.php(92): locate_template(Array, true, true, Array)
#13 /home/customer/www/c42.studio/public_html/wp-content/themes/hello-elementor/index.php(37): get_footer()
#14 /home/customer/www/c42.studio/public_html/wp-includes/template-loader.php(106): include('/home/customer/...')
#15 /home/customer/www/c42.studio/public_html/wp-blog-header.php(19): require_once('/home/customer/...')
#16 /home/customer/www/c42.studio/public_html/index.php(17): require('/home/customer/...')
#17 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 9 of 9JS: 2023-03-12 18:10:57 [error X 12][https://c42.studio/wp-admin/load-scripts.php?c=0&loadchunk_0=underscore,jquery-core,jquery-migrate,backbone,jquery-ui-core,jquery-ui-mouse,wp-api-request,regenerator-runtime,wp-polyfill,wp-&loadchunk_1=hooks&ver=6.1.1:4:51824] Cannot read properties of undefined (reading 'replace') 
JS: 2023-03-12 23:41:12 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view6699") has already been destroyed and cannot be used. 
JS: 2023-03-12 23:43:04 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view21175") has already been destroyed and cannot be used. 
JS: 2023-03-12 23:44:01 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view35378") has already been destroyed and cannot be used. 
JS: 2023-03-12 23:51:37 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4920") has already been destroyed and cannot be used. 
JS: 2023-03-13 11:04:03 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view12656") has already been destroyed and cannot be used. 
JS: 2023-03-13 13:34:13 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view10755") has already been destroyed and cannot be used. 
JS: 2023-03-13 13:34:24 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view12957") has already been destroyed and cannot be used. 
JS: 2023-03-16 23:52:31 [error X 1][https://c42.studio/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.5:3:791161] Cannot read properties of undefined (reading 'toLowerCase') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Save UTMs to Elementor Form: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by black-eye** [comment]
Oh yes, welcome to the club: https://github.com/elementor/elementor/issues/16819
The problem is caused by the custom breakpoints. If you can, turn them off.

**Comment by Irinel123** [comment]
+1
I have the same problem. I notice the problem is from 3.12.0 (for 3.11.5 it looks fine)

Do you have solutions?

**Comment by TomRaraAvis** [comment]
I had this issue as well. I worked for **many hours** trying to figure out WTF was happening. Finally, after reading comments in #16819 I regenerated CSS & Data, cleared my cache, and all is well. Elementor > Tools > General > Regenerate CSS & Data. After all that frustration, one click and fixed. It's worth a shot. Maybe it works for you, maybe it doesn't. Please report back so others know.

<img width="509" alt="regenerate-css" src="https://user-images.githubusercontent.com/1435355/229851745-b41964c3-13b6-417c-a380-6969154ee1cf.png">

**Comment by lensbreak** [comment]
+1

**Comment by Nevaah** [comment]
> I had this issue as well. I worked for **many hours** trying to figure out WTF was happening. Finally, after reading comments in #16819 I regenerated CSS & Data, cleared my cache, and all is well. Elementor > Tools > General > Regenerate CSS & Data. After all that frustration, one click and fixed. It's worth a shot. Maybe it works for you, maybe it doesn't. Please report back so others know.
> 
> <img alt="regenerate-css" width="509" src="https://user-images.githubusercontent.com/1435355/229851745-b41964c3-13b6-417c-a380-6969154ee1cf.png">

This is a temp fix and works until after next edit. I currently have a live site where client demands the extra breakpoints. I can scrape by this way, but this is a terrible stain on Elementor to leave a known issue open for 18 months.

**Comment by up-creator** [comment]
I feel like this bug is random, then on one site it works as usual, then on another site it works if i regenerate, on a third site, it started working after i set a column to 100% on mobile and then back to 50%, even though the bug existed on other columns not edited. We only have the basic tablet and mobile(default), no custom breakpoints set.

**Comment by black-eye** [comment]
> I feel like this bug is random, then on one site it works as usual, then on another site it works if i regenerate, on a third site, it started working after i set a column to 100% on mobile and then back to 50%, even though the bug existed on other columns not edited. We only have the basic tablet and mobile(default), no custom breakpoints set.

Just to make it clear once again - this issue doesn't depend on wheather you have a custom breakpoint set or not, It depends on wheather you have the Custom breakpoint feature activated or not: 

![obrazek](https://user-images.githubusercontent.com/5634870/232488558-1df8f468-0b36-4856-8894-0c9b205a396b.png)

Please check out, wheather you have this experiment on. And also note, that on version 3.12 the version was turn on by default - so you didn't have to turn it on yourself. Hope it helps.

**Comment by up-creator** [comment]
> > I feel like this bug is random, then on one site it works as usual, then on another site it works if i regenerate, on a third site, it started working after i set a column to 100% on mobile and then back to 50%, even though the bug existed on other columns not edited. We only have the basic tablet and mobile(default), no custom breakpoints set.
> 
> Just to make it clear once again - this issue doesn't depend on wheather you have a custom breakpoint set or not, It depends on wheather you have the Custom breakpoint feature activated or not:
> 
> ![obrazek](https://user-images.githubusercontent.com/5634870/232488558-1df8f468-0b36-4856-8894-0c9b205a396b.png)
> 
> Please check out, wheather you have this experiment on. And also note, that on version 3.12 the version was turn on by default - so you didn't have to turn it on yourself. Hope it helps.

Hi black-eye

Sorry, i misunderstood the custom breakpoints.
It's is working now, if turned off, thank you!

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core 3.15.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
